RuntimeDetour/Platforms/Runtime/DetourRuntimeNETCorePlatform.cs

```diff
@@ -5,20 +5,144 @@
+
+        protected static IntPtr GetJitObject() {
+            if (getJit == null) {
+                // To make sure we get the right clrjit, we enumerate the process's modules and find the one 
+                //   with the name we care aboutm, then use its full path to gat a handle and load symbols.
+                Process currentProc = Process.GetCurrentProcess();
+                ProcessModule clrjitModule = currentProc.Modules.Cast<ProcessModule>()
+                    .FirstOrDefault(m => Path.GetFileNameWithoutExtension(m.FileName).EndsWith("clrjit"));
+                if (clrjitModule == null)
+                    throw new PlatformNotSupportedException();
+
+                if (!DynDll.TryOpenLibrary(clrjitModule.FileName, out IntPtr clrjitPtr))
+                    throw new PlatformNotSupportedException();
+
+                if (PlatformHelper.Is(Platform.Windows)) {
+                    // we can use this chekc only on Windows, because only Windows actually has FilveVersionInfo
+                    // this is preferred because it checks the version of the JIT, rather than the runtime library
+                    //   which is what actually determines the layout.
+                    isNet5Jit = clrjitModule.FileVersionInfo.ProductMajorPart >= 5;
+                } else {
+                    // this gets System.Private.CorLib's major version, which *should* match the runtime version
+                    // it is the only method we have at the moment to detect it on non-Windows platforms
+                    isNet5Jit = typeof(object).Assembly.GetName().Version.Major >= 5;
+                }
+
+                try {
+                    getJit = clrjitPtr.GetFunction(nameof(getJit)).AsDelegate<d_getJit>();
+                } catch {
+                    DynDll.CloseLibrary(clrjitPtr);
+                    throw;
+                }
+            }
+
+            return getJit();
+        }
+
+        private static bool isNet5Jit;
+
+        protected static Guid GetJitGuid(IntPtr jit) {
+            int getVersionIdentIndex = isNet5Jit ? vtableIndex_ICorJitCompiler_getVersionIdentifier_net5
+                                                 : vtableIndex_ICorJitCompiler_getVersionIdentifier;
+            d_getVersionIdentifier getVersionIdentifier = ReadObjectVTable(jit, getVersionIdentIndex)
+                .AsDelegate<d_getVersionIdentifier>();
+            getVersionIdentifier(jit, out Guid guid);
+            return guid;
+        }
+
+        // The offset to use is determined in GetJitObject where other properties of the JIT are determined
+        private const int vtableIndex_ICorJitCompiler_getVersionIdentifier = 4;
+        private const int vtableIndex_ICorJitCompiler_getVersionIdentifier_net5 = 2;
+        [UnmanagedFunctionPointer(CallingConvention.ThisCall)]
+        private delegate void d_getVersionIdentifier(
+            IntPtr thisPtr, // ICorJitCompiler*
+            out Guid versionIdentifier
+            );
```

Comment:
That ); looks off to me - is it supposed to be that way?
```suggestion
        private delegate void d_getVersionIdentifier(
            IntPtr thisPtr, // ICorJitCompiler*
            out Guid versionIdentifier
        );
```

Reply:
Not particularly. I'll make the change.